doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+
+An instruction category is sampled at the ID/EX stage (which is where all the varying behaviours actually occur).
+Some categories are just a single instruction which is named without further description.
+
+
+* ALU - All of the reg/reg reg/imm instructions that use the ALU.
+  This is any RISC-V instruction with an opcode of ``7’b0010011`` or ``7’b0110011`` (``ibex_pkg::OPCODE_OP`` and ``ibex_pkg::OPCODE_OP_IMM``) other than the ``MUL*`` and ``DIV*`` family of instructions (from RV32M).
+* Mul - Any ``MUL*`` instruction (from RV32M).
+* Div - Any ``DIV*`` instruction (from RV32M).
+* Branch - Any ``B*`` family branch instruction.
+* Jump - ``JAL``/``JALR``
+* Load - Any ``L*`` family load instruction.
+* Store - Any ``S*`` family load instruction.
+* CSRAccess - Any instruction from Zicsr.
+* EBreakDbg/EBreakExc- An ``EBREAK`` instruction that either enters debug mode (Dbg) or causes an exception (Exc).
+  Which occurs depends upon the setting of ``dcsr.ebreakm`` / ``dcsr.ebreaku`` combined with the privilege level of executed instruction.
+* ``ECALL``
+* ``MRET``
+* ``DRET``
+* ``WFI``
+* ``FENCE.I``
+* FetchError - Any instruction that saw a fetch error.
+* CompressedIllegal - Any compressed instruction with an illegal encoding.
+* UncompressedIllegal - Any uncompressed instruction with an illegal encoding.
+* CSRIllegal - Any instruction attempting a CSR access that is not allowed.
+* PrivIllegal - Illegal due to privilege level or being in/out of debug mode.
+* OtherIllegal - Any other instruction that raises an Illegal instruction exception that isn’t in the other categories.
+* None - No instruction in ID/EX stage.
+
+Stalls
+""""""
+Not all instructions can see all kinds of stalls.
+A stall category is sampled at the ID/EX stage only (as stalls in IF and WB don’t break down into categories).
+
+* Instr - A stall cause by a multi-cycle instruction.
+  This can be seen by instructions from categories:
+
+  * MUL
+  * DIV
+  * Branch
+  * Jump
+
+* LdHz - A load hazard, the instruction in ID/EX depends upon the result of a load that is awaiting a response in writeback.
+  This can be seen by instructions from categories:
+
+  * ALU
+  * Mul
+  * Div
+  * Branch
+  * Jump
+  * Load
+  * Store
+  * CSRAccess
+
+* Mem - Memory stall, the instruction in ID/EX is awaiting a prior memory request to complete before it can begin (to allow precise interrupts on a memory error response). This can be seen for all instruction categories
+
+Privilege Level
+"""""""""""""""
+
+Ibex can operate at either the M (machine) or U (user) privilege levels.
+Different aspects of the Ibex microarchitecture can be using different privilege levels at once.
+
+* Privilege level of IF stage instruction.
+* Privilege level of ID/EX stage instruction.
+* Privilege level of LSU operation (ID/EX privilege level modified by `mstatus.mprv` and `mstatus.mpp` settings).
+
+Hazards
+"""""""
+Ibex hazards all occur in the interaction between the ID and EX stage.
```

Comment:
Nit: You've added an extra line after headings above here. I probably prefer this more condensed style, but it's maybe a good idea to be consistent.